### ISSUE TITLE
Use `select` instead of `pattern` in `data_rename()`?

### ISSUE THREAD

**Issue by strengejacke** [easystats/datawizard#566]
I think when the function was initially written, it was very feature-limited. That's why we had the argument `pattern`, instead of `select`. Meanwhile, `data_rename()` is much more flexible, and the `pattern` argument came closer to what `select` typically does or can do.

Should we rename `pattern` into `select` then? Opinions?

@easystats/core-team @etiennebacher @DominiqueMakowski

**Comment by etiennebacher** [comment]
I agree, `select` seems more appropriate and in line with the rest of the package.

**Comment by bwiernik** [comment]
Could you elaborate on what the argument does and why you think select is a better name?

**Comment by strengejacke** [comment]
`data_rename()` renames colums, where `pattern` _selects_ the columns to rename, and `rename` contains the new names. In a very early version, this was very simple, you needed to specify column names as character vector for `pattern` and the same amount of new names as character vector for `rename`.

So, the primary task of `pattern` is to _select_ variables.

Meanwhile, `data_rename()` is more flexible. We first startet adding options to directly rename variables in `pattern`, e.g.:

```r
data_rename(iris, pattern = c(NewSpec = "Species"))
```

You could do the same with `data_select()`, actually, because the `select` argument is very flexible:

```r
data_select(iris, select = c(NewSpec = "Species"))
```

Furthermore, the `rename` argument is also more flexible, as it now can include tokens. If you want to select all columns starting with "old_" in their name, you currently have to write them all as character vector:

```r
data_rename(data, pattern = c("old_1", "old_2", "old_", ...))
```

But if a) `pattern` only selects variables, and b) `rename` is more flexible now, why don't we use out `select` functionality in `data_rename()` as well? You could then do following to rename a bunch of columns that follow a specific pattern:

```r
data_rename(data, select = starts_with("old_"), rename = "new_{letter}")
```

I think there's no need to limit the functionality of selecting variables in `data_rename()`. Sure, we have `data_select()`, so we could do:

```r
data_select(data, select = starts_with("old_")) |> data_rename(rename = "new_{letter}")
```

as `data_rename()` would rename _all_ columns if `pattern` is not provided, but this approach returns only the selected columns, while the other example where you directly `select` in `data_rename()` returns the full data frame, with new names for the selected variables only.

**Comment by bwiernik** [comment]
Yeah, I think we should switch to `select`
